v2/tlsworkload/workload.go

```diff
@@ -0,0 +1,70 @@
+package tlsworkload
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/spiffex509"
+)
+
+// ErrNotReady is returned when the workload has yet to receive the initial
+// response from the Workload API.
+var ErrNotReady = errors.New("tls-workload: not ready")
+
+// ErrClosed is returned from methods that provide up-to-date information from
+// the Workload API after the workload has been closed to prevent callers from
+// relying on stale information.
+var ErrClosed = errors.New("tls-workload: closed")
+
+// Workload is a TLS workload that maintains up-to-date TLS context (i.e.
+// SVIDs with private keys, trusted X.509 roots, etc.) retrieved from the
+// Workload API.
+type Workload struct{}
+
+// Open opens the TLS workload against the Workload API. It does not return
+// until the workload has received the first update from the Workload API,
+// unless the NoWait option is used.
+func Open(ctx context.Context, options ...Option) (*Workload, error) {
+	panic("not implemented")
+}
+
+// Close closes the TLS workload, tearing down streams and disconnecting
+// from the workload API. Other methods that rely on up-to-date information
+// from the workload API will fail with ErrClosed after this call to prevent
+// callers from relying on stale information.
+func (w *Workload) Close() error {
```

Comment:
it's likely that `Close()` may need to accept a context too

Reply:
I'm not so sure. Internally, I foresee these workloads having a grpc.ClientConn open to the Workload API with one or more open streams. Neither cancelling the context on the streams or closing the grpc.ClientConn are blocking operations AFAIK, and even if so, do not provide a way to thread through a context (i.e. aren't themselves cancelable).